avrohugger-core/src/main/scala/input/parsers/FileInputParser.scala

```diff
@@ -89,8 +97,8 @@ class FileInputParser {
           val protocol = Protocol.parse(infile)
           List(Right(protocol))
         case "avdl" =>
-          val idlParser = new Idl(infile, classLoader)
-          val protocol = idlParser.CompilationUnit()
+          val idl = new IdlReader().parse(infile.toPath())
```

Comment:
we need a parse method that accepts our `classLoader`, as was available for the `Idl` constructor